include/nlohmann/json.hpp

```diff
@@ -5697,17 +5807,27 @@ class basic_json
         // insert only works for arrays
         if (JSON_HEDLEY_UNLIKELY(!is_array()))
         {
-            JSON_THROW(type_error::create(309, "cannot use insert() with " + std::string(type_name())));
+            JSON_THROW(type_error::create(309, "cannot use insert() with " + std::string(type_name()), diagnostics_t(*this)));
         }
 
         // check if iterator pos fits to this JSON value
         if (JSON_HEDLEY_UNLIKELY(pos.m_object != this))
         {
-            JSON_THROW(invalid_iterator::create(202, "iterator does not fit current value"));
+            JSON_THROW(invalid_iterator::create(202, "iterator does not fit current value", diagnostics_t(*this)));
         }
 
         // insert to array and return iterator
+#if JSON_DIAGNOSTICS
+        const auto size = ilist.size();
+        iterator result =  insert_iterator(pos, ilist.begin(), ilist.end());
+        for (std::size_t i = 0; i < size; ++i)
+        {
+            (result + static_cast<typename iterator::difference_type>(i))->m_parent = this;
+        }
+        return result;
```

Comment:
As above

        iterator result =  insert_iterator(pos, ilist.begin(), ilist.end());
    #if JSON_DIAGNOSTICS
        for (typename iterator::difference_type i = 0; i < static_cast<typename iterator::difference_type>(ilist.size()); ++i)
        {
            set_parent(result + i);
        }
    #endif
        return result;

Reply:
Fixed with f8037660d06a0fbb6b1cde944d436032d04564ae.